modin/pandas/base.py

```diff
@@ -1953,12 +1948,19 @@ def pow(
             "pow", other, axis=axis, level=level, fill_value=fill_value
         )
 
-    def quantile(
-        self, q=0.5, axis=0, numeric_only=True, interpolation="linear"
+    def _quantile(
+        self, q, axis, numeric_only, interpolation, method
     ):  # noqa: PR01, RT01, D200
         """
         Return values at the given quantile over requested axis.
         """
+        if PandasCompatVersion.CURRENT != PandasCompatVersion.LATEST:
+            if method != "single":
+                raise ValueError(f"Unsupported method={method} for quantile")
```

Comment:
Should we raise the error in `Python36CompatibleBasePandasDataset.quantile`?

Reply:
actually, this should be an assert - there should not be a user-facing way to pass different `method` value in compat mode

Reply:
changed to `assert`